src/revlanguage/moves/Move_IndependentPriorSampler.h

```diff
@@ -0,0 +1,211 @@
+    return new Move_IndependentPriorSampler<rlValueType>(*this);
+}
+
+
+template <class rlValueType>
+void RevLanguage::Move_IndependentPriorSampler<rlValueType>::constructInternalObject( void )
+{
+    // we free the memory first
+    delete value;
+    
+    // now allocate a new independent prior sampler move
+    double w = static_cast<const RealPos &>( weight->getRevObject() ).getValue();
+    
+    RevBayesCore::TypedDagNode<typename rlValueType::valueType>* tmp = static_cast<const rlValueType &>( x->getRevObject() ).getDagNode();
+    RevBayesCore::StochasticNode< typename rlValueType::valueType > *sn = static_cast<RevBayesCore::StochasticNode< typename rlValueType::valueType > *>( tmp );
+    
+    bool mh = static_cast<const RlBoolean &>( metropolisHastings->getRevObject() ).getValue();
+
+    RevBayesCore::Proposal *p = new RevBayesCore::IndependentPriorProposal< typename rlValueType::valueType >(sn, mh);
+    value = new RevBayesCore::MetropolisHastingsMove(p, w, false);
+}
+
+
+/** Get Rev type of object */
+template <class rlValueType>
+const std::string& RevLanguage::Move_IndependentPriorSampler<rlValueType>::getClassType(void)
+{
+    
+//    static std::string rev_type = "Move_IndependentPriorSampler<" + rlValueType::getClassType() + ">";
+    static std::string rev_type = "Move_IndependentPriorSampler__" + rlValueType::getClassType();
+    
+    return rev_type;
+}
+
+/** Get class type spec describing type of object */
+template <class rlValueType>
+const RevLanguage::TypeSpec& RevLanguage::Move_IndependentPriorSampler<rlValueType>::getClassTypeSpec(void)
+{
+    
+    static TypeSpec rev_type_spec = TypeSpec( getClassType(), new TypeSpec( Move::getClassTypeSpec() ) );
+    
+    return rev_type_spec;
+}
+
+
+/**
+ * Get the Rev name for the constructor function.
+ *
+ * \return Rev name of constructor function.
+ */
+template <class rlValueType>
+std::string RevLanguage::Move_IndependentPriorSampler<rlValueType>::getMoveName( void ) const
+{
+    // create a constructor function name variable that is the same for all instance of this class
+    std::string c_name = "IidPrior";
+    
+    return c_name;
+}
+
+
+/** Return member rules (no members) */
+template <class rlValueType>
+const RevLanguage::MemberRules& RevLanguage::Move_IndependentPriorSampler<rlValueType>::getParameterRules(void) const
+{
+    
+    static MemberRules move_member_rules;
+    static bool rules_set = false;
+    
+    if ( !rules_set )
+    {
+        move_member_rules.push_back( new ArgumentRule( "x", rlValueType::getClassTypeSpec(), "The variable on which this move operates.", ArgumentRule::BY_REFERENCE, ArgumentRule::STOCHASTIC ) );
+        move_member_rules.push_back( new ArgumentRule( "MetropolisHastings"   , RlBoolean::getClassTypeSpec(), "If TRUE, use the regular Metropolis-Hastings acceptance ratio. If FALSE, always accept this move and sample every variable uniformly.", ArgumentRule::BY_VALUE, ArgumentRule::ANY, new RlBoolean( true ) ) );
```

Comment:
Can you explain why someone would want this to be false?

My impression is that this move proposes a new value from the prior.  If that is the case, then always accepting is equivalent to ignoring likelihoods that condition on this node.

Reply:
If I remember correctly, that was used to show that the incorrect version can give biased estimates. Don't remember when or why we even tested that (@jsigao). We can remove this option for the released version if you prefer.

Reply:
Ah, I see.  Let's remove it for the released version though.